src/Metadata/Resource/ResourceMetadataCollection.php

```diff
@@ -86,6 +88,11 @@ public function getOperation(?string $operationName = null, bool $forceCollectio
         $this->handleNotFound($operationName, $metadata);
     }
 
+    public function isDynamic(): bool
+    {
+        return str_starts_with($this->resourceClass, self::DYNAMIC_RESOURCE_CLASS_PREFIX);
+    }
```

Comment:
I really don't like this graphql-specific code here, can you explain this?

Reply:
This is not GraphQL-specific. The aim is to create metadata for a resource which doesn't exist.